docs/docs/Deployment/deployment-public-server.md

```diff
@@ -0,0 +1,147 @@
+However, you can forward Langflow server traffic with a forwarding platform like [ngrok](https://ngrok.com/docs/getting-started/) or [zrok](https://docs.zrok.io/docs/getting-started) to make your server public.
+
+This allows you to [deploy your MCP server externally](#deploy-your-mcp-server-externally), [serve API requests](#serve-api-requests), and [share your playground externally](#share-your-playground-externally).
+
+The following procedure uses ngrok, but you can use any similar reverse proxy or forwarding platform.
+This procedure also assumes that you're using the default Langflow listening address `http://localhost:7860`.
+
+## Prerequisites
+
+- [A running Langflow instance](/get-started-installation)
+- [ngrok installed](https://ngrok.com/docs/getting-started/#1-install-ngrok)
+- [An ngrok authtoken](https://dashboard.ngrok.com/get-started/your-authtoken)
+
+## Expose your Langflow server with ngrok
+
+1. Start Langflow.
+
+    ```bash
+    uv run langflow run
+    ```
+
+2. In another terminal window, copy your [ngrok authtoken](https://dashboard.ngrok.com/get-started/your-authtoken) and use it to authenticate your local ngrok server:
+
+    ```bash
+    ngrok config add-authtoken NGROK_TOKEN
+    ```
+
+    Replace `NGROK_TOKEN` with your ngrok authtoken.
+
+3. Use ngrok to expose your Langflow server to the public internet:
+
+    ```bash
+    ngrok http http://localhost:7860
+    ```
+
+    The ngrok session starts in your terminal and deploys an ephemeral domain with no authentication.
+    To add authentication or deploy a static domain, see the [ngrok documentation](https://ngrok.com/docs/).
+
+
+    The `Forwarding` row displays the forwarding address for your Langflow server:
+
+    ```
+    Forwarding https://94b1-76-64-171-14.ngrok-free.app -> http://localhost:7860
+    ```
+
+    The forwarding address is acting as a reverse proxy for your Langflow server.
+
+4. Open the URL where ngrok is forwarding your local traffic, such as `https://94b1-76-64-171-14.ngrok-free.app`.
+
+    Your Langflow server is now publicly available at this domain.
+
+## Deploy your MCP server externally
+
+To deploy your Langflow MCP server with ngrok, do the following:
+
+1. Select the Langflow project that contains the flows you want to serve as tools, and then click the **MCP Server** tab.
+
+    Note that the code template now contains your ngrok forwarding address instead of the localhost address.
+    ```json
+    {
+      "mcpServers": {
+        "PROJECT_NAME": {
+          "command": "uvx",
+          "args": [
+            "mcp-proxy",
+            "https://3f7c-73-64-93-151.ngrok-free.app/api/v1/mcp/project/d764c4b8-5cec-4c0f-9de0-4b419b11901a/sse"
+          ]
+        }
+      }
+    }
+    ```
+
+2. Complete the steps in [Connect clients to Langflow's MCP server](/mcp-server#connect-clients-to-use-the-servers-actions) using the ngrok forwarding address.
```

Comment:
Actually, this code snippet and these steps are basically all already in the _connect-clients-to-use-the-servers-action_s section. I think we could streamline the content between this section and the MCP server page. It's a bit back-and-forth right now.

We can look at it together next week, if you want.

Reply:
Or if you want to get this published, we can revisit it in another PR. I have to make changes to MCP for the welcome page anyway.